### DISCUSSION TITLE
Deprecate device tracker battery level attribute

### DISCUSSION THREAD

**Discussion by MartinHjelmare** [home-assistant/architecture#627] [Entity Models] [answered]
## Proposal

I suggest we deprecate the `battery_level` state attribute for non legacy device tracker entities, ScannerEntity and TrackerEntity.

## Background

The design of these entities were made before we started breaking out battery level measurements to separate sensor entities. Our current design standard says that a battery level measurement should be represented by a separate sensor entity with device class battery.

https://developers.home-assistant.io/docs/core/entity/device-tracker

https://developers.home-assistant.io/docs/core/entity/sensor#available-device-classes

## Scope

Affected core integrations:

- gpslogger
- icloud
- mobile_app
- owntracks
- starline
- traccar
- tractive
- zha

Affected custom integrations:

?

## Solution

- The battery level state attribute in the `state_attributes` property of the base class `BaseTrackerEntity` will be removed.
- The `battery_level` property in the `ScannerEntity` and `TrackerEntity` entities should be marked as deprecated and log a warning if used for a deprecation period. The deprecation period is done for custom integration authors.
- Affected integrations should be migrated by removing the `battery_level` property in `ScannerEntity` and `TrackerEntity` entities and replace it with a sensor platform with an entity with device class battery.

**Comment by dgomes** [comment]
mqtt integrations (such as z2m) are also affected

**Reply by MartinHjelmare** [reply]
I don't find where the mqtt integration sets the `battery_level` property for its `TrackerEntity` class definition. Do you have a link?

Note that we won't stop custom integrations from setting battery level as an extra state attribute. That's not part of the scope of this proposal.

**Reply by dgomes** [reply]
My bad, I understand now that this applies only to device_trackers, so no issue identified

**Reply by emontnemery** [reply]
Did you refer to how mqtt allows you to write any attribute?

**Reply by dgomes** [reply]
I just recalled that z2m entities have battery_level attribute

but since none of its entities are device_tracker's....

**Comment by frenck** [comment, marked as answer]
I think we should deprecate this; we are heading toward being 2 years after this architecture proposal; and this nowadays most certainly doesn't first our current design.

👍 

../Frenck
